prow/spyglass/artifactfetchers.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spyglass
+
+import (
+	"errors"
+
+	"k8s.io/test-infra/prow/spyglass/viewers"
+)
+
+var (
+	ErrCannotParseSource = errors.New("could not create job source from provided source")
+)
+
+// JobSource gets information about a location storing the results of a single Prow job
+type JobSource interface {
+	// CanonicalLink returns a link to the storage location this job's artifacts.
+	CanonicalLink() string
+	// JobPath returns the common prefix (excluding the name of the bucket) of all artifacts produced
+	// by this job
+	JobPath() string
+	BucketName() string
```

Comment:
For example, this seems an especially GCS-centric abstraction for an interface.

I also dislike the idea that BucketName should be treated separately from the rest of the URI anywhere aside from the immediate interaction with GCS. Working with unix's `/foo/bar` is substantially nicer than windows' `c:\bar`

Reply:
I don't think bucket name should be exported at all.

Reply:
It's not clear why this type is exported at all? Seems like we should have a local implementation of `Artifact` which has a reasonable public API, and other details should be private / opaque...

Reply:
Removing `BucketName` method and unexporting `JobSource`